src/ImageSharp.Web/DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -26,55 +24,26 @@ public static class ServiceCollectionExtensions
         /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
         /// <returns>An <see cref="IImageSharpBuilder"/> that can be used to further configure the ImageSharp services.</returns>
         public static IImageSharpBuilder AddImageSharp(this IServiceCollection services)
-            => AddImageSharp(services, _ => { });
+            => new ImageSharpBuilder(services).AddDefaultServices();
 
         /// <summary>
         /// Adds ImageSharp services to the specified <see cref="IServiceCollection" /> with the given options.
         /// </summary>
         /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
-        /// <param name="setupAction">An <see cref="Action{ImageSharpMiddlewareOptions}"/> to configure the provided <see cref="ImageSharpMiddlewareOptions"/>.</param>
+        /// <param name="configureOptions">An <see cref="Action{ImageSharpMiddlewareOptions}"/> to configure the provided <see cref="ImageSharpMiddlewareOptions"/>.</param>
         /// <returns>An <see cref="IImageSharpBuilder"/> that can be used to further configure the ImageSharp services.</returns>
-        public static IImageSharpBuilder AddImageSharp(
-            this IServiceCollection services,
-            Action<ImageSharpMiddlewareOptions> setupAction)
-        {
-            Guard.NotNull(services, nameof(services));
-            Guard.NotNull(setupAction, nameof(setupAction));
-
-            services.TryAddTransient<IConfigureOptions<ImageSharpMiddlewareOptions>, ImageSharpConfiguration>();
-
-            IImageSharpBuilder builder = new ImageSharpBuilder(services);
+        public static IImageSharpBuilder AddImageSharp(this IServiceCollection services, Action<ImageSharpMiddlewareOptions> configureOptions)
+            => services.AddImageSharp().Configure(configureOptions);
 
-            AddDefaultServices(builder, setupAction);
-
-            return builder;
-        }
-
-        private static void AddDefaultServices(
-            IImageSharpBuilder builder,
-            Action<ImageSharpMiddlewareOptions> setupAction)
+        private static IImageSharpBuilder AddDefaultServices(this IImageSharpBuilder builder)
         {
-            builder.Services.Configure(setupAction);
```

Comment:
One thing this call did was internally call [`AddOptions()`](https://github.com/dotnet/runtime/blob/bf0492f520c2530c4cc475849e74bf55541e52b8/src/libraries/Microsoft.Extensions.Options/src/OptionsServiceCollectionExtensions.cs#L22-L30), so the generic options interfaces are registered and can be resolved from the service container.

Looking at other extension methods in [ASP.NET Core](https://github.com/dotnet/aspnetcore), they do explicitly call `AddOptions()` as well, so let's make sure ImageSharp.Web does the same (even if it's _probably_ already added anyway, we can't be sure though).

Reply:
Fixed in https://github.com/SixLabors/ImageSharp.Web/pull/245/commits/fbb6cef1eb6affc8ad470aea252c8f459e886536:
https://github.com/SixLabors/ImageSharp.Web/blob/fbb6cef1eb6affc8ad470aea252c8f459e886536/src/ImageSharp.Web/DependencyInjection/ServiceCollectionExtensions.cs#L40

Reply:
I'm sorry but I simply do not follow. Why are you changing any of this? We're only supposed to be removing the no-op `ImageSharpConfiguration`